### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by larister**
Depends on https://github.com/bw-allardvanhelbergen/go/pull/56 and will need a rebase once that's merged.

This took a little longer than hoped, but it should all work fine now. I'm not happy with having to use `util.inspects` when logging out the mongo model, as winston should [do that itself](https://github.com/winstonjs/winston#logging-with-metadata). Still, it works; happy to add a TODO if you want.

Mark E also recommended using Bunyan instead; I ran out of time to implement that but feel free to give it a look if you want.

### CODE REVIEW

File: lib/logging.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+var NODE_ENV = (process.env.NODE_ENV || '').toLowerCase();
+
+winston.remove(winston.transports.Console);
+
+function init() {
+    var loggingDir;
+
+    switch (NODE_ENV) {
+    case 'production':
+        production = true;
```
Comment (bw-allardvanhelbergen) [inline lib/logging.js]: The `var production` looks superfluous. Is it used for anything?

File: lib/logging.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        loggingDir = path.resolve(__dirname, config.logging.DIR);
+        console.log('Enabling logging to ', loggingDir);
+        winston.add(winston.transports.File, {
+            stream: new LogStream({
+                logdir: loggingDir,
+                nameformat: '[brandwatch.]YYYY-MM-DD[.log]',
```
Comment (bw-allardvanhelbergen) [inline lib/logging.js]: There's multiple apps running on lightning. Perhaps [go.] instead of [brandwatch.] would be better?

**Comment by bw-allardvanhelbergen** [comment]
Oh, awesome! Thank you so much for this PR @larister! Really excited about this. :)

There's a nice comparison of [Winston VS Bunyan here](https://strongloop.com/strongblog/compare-node-js-logging-winston-bunyan/). Bunyan definitely looks more advanced and I'm happy to use Bunyan if you think that's better.

For now I'm super happy to start with this do any kind of logging at all. :)

File: lib/logging.js (hunk 1 above)
Comment (larister) [inline lib/logging.js]: Great spot; in Frontend this is exported but I'm not a big fan of that pattern so I removed the export, forgetting to remove the var :D

File: lib/logging.js (hunk 2 above)
Comment (larister) [inline lib/logging.js]: In the production config, the log dir is specified as being within the /go dir, but I think you're right anyway, better safe than sorry.

**Comment by larister** [comment]
Agreed, let's leave Bunyan for a later stage

**Comment by bw-allardvanhelbergen** [comment]
@larister, I merged the #56. Feel free to rebase this when you get the chance.

**Comment by larister** [comment]
Thanks @bw-allardvanhelbergen, rebased & pushed.

**Pull request merged**